crates/flux-refineck/src/lib.rs

```diff
@@ -112,15 +112,22 @@ pub fn check_fn(
         let ghost_stmts = compute_ghost_statements(genv, def_id)
             .with_span(span)
             .map_err(|err| err.emit(genv, def_id))?;
-
+        let mut closures = UnordMap::default();
```

Comment:
Oh, I didn't notice this the first time. We want to use a different map for each phase.